doc/wallet-storage.md

```diff
@@ -0,0 +1,255 @@
+# Wallet Storage Design
+
+In current state libindy allows to plug different wallet implementations. Plugged wallet now handles both security
+and storage layers. This design proposes to restrict our plugged interface by handling only storage layer.
+All encryption will be performed in libindy. It will simplify plugged wallets and warranty good security level
+for 3d party wallets implementations.
+
+Also proposals enhances our API for efficient and flexible search with paging support.
+
+## Goals and ideas
+
+* Simplify plugged wallets and warranty security level for 3d party wallets implementation
+* Perform all encryption on libindy level:
+  * Entity ids will be always encrypted
+  * Entity values will be always encrypted
+  * Entity metadata will be always encrypted
+  * Tag names will be always encrypted
+  * Tag values will be **optionally** encrypted. If user wants to perform some complex searches it can be possible
+    to include some additional un-encrypted tags
+* Allow plugging of different storages
```

Comment:
Do we consider that Wallet interface is public, so that the App can set/get/addTags to the wallet?